sdl_viewer/src/lib.rs

```diff
@@ -106,28 +109,21 @@ impl SdlViewer {
 
         // call octree generation functions
         let mut octree_opt: Option<Box<Octree>> = None;
-        for (prefix, octree_factory_function) in &self.octree_factory_functions {
-            if !octree_path.starts_with(prefix) {
+        for (prefix, octree_factory_function) in &self.octree_factories {
+            if !octree_argument.starts_with(prefix) {
                 continue;
             }
-            let mut no_prefix_path = String::from(octree_path);
-            no_prefix_path.drain(..prefix.len());
-            if let Ok(o) = octree_factory_function(&no_prefix_path) {
+            let no_prefix = &octree_argument[prefix.len()..];
```

Comment:
use to_string here directly? I prefer to_string() a bit because it makes it clear what is created, compared to into() which just gives the information: "whatever the callsite wants"